bootstrap/pkg/apis/apps/configconverters/converters.go

```diff
@@ -26,21 +24,9 @@ const (
 	Api = "kfdef.apps.kubeflow.org"
 )
 
-func isValidUrl(toTest string) bool {
-	_, err := netUrl.ParseRequestURI(toTest)
-	if err != nil {
-		return false
-	} else {
-		return true
-	}
-}
-
-// LoadConfigFromURI reads the kfdef from a remote URI or local file,
-// and returns the kfconfig.
-// It will set the AppDir and ConfigFilename in kfconfig:
-//   AppDir = cwd if configFile is remote, or it will be the dir of configFile.
-//   ConfigFilename = the file name of configFile.
-func LoadConfigFromURI(configFile string) (*kfconfig.KfConfig, error) {
+// This function should be thread safe and should not write to disk.
+// LoadFileFromURI: get file content from URI (lcoal or remote) and return as []byte
+func LoadFileFromURI(configFile string) ([]byte, error) {
```

Comment:
Why are we returning things as []byte? If we want to return a variable type couldn't we do interface{} and then the caller can do a type assertion?

Reply:
The converter interface now takes file content in []byte as input:
https://github.com/kubeflow/kubeflow/blob/master/bootstrap/pkg/apis/apps/configconverters/converters.go#L22
If use interface{} then need to convert it back later